device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSenderLinkHandler.java

```diff
@@ -273,13 +280,22 @@ MessageImpl iotHubMessageToProtonMessage(Message message)
             userProperties.put(MessageProperty.IOTHUB_CREATION_TIME_UTC, message.getCreationTimeUTCString());
         }
 
+        ApplicationProperties applicationProperties = new ApplicationProperties(userProperties);
+        outgoingMessage.setApplicationProperties(applicationProperties);
+
+        Map<Symbol, Object> messageAnnotationsMap = new HashMap<>();
         if (message.isSecurityMessage())
         {
-            userProperties.put(MessageProperty.IOTHUB_SECURITY_INTERFACE_ID, MessageProperty.IOTHUB_SECURITY_INTERFACE_ID_VALUE);
+            messageAnnotationsMap.put(Symbol.valueOf(MessageProperty.IOTHUB_SECURITY_INTERFACE_ID), MessageProperty.IOTHUB_SECURITY_INTERFACE_ID_VALUE);
         }
 
-        ApplicationProperties applicationProperties = new ApplicationProperties(userProperties);
-        outgoingMessage.setApplicationProperties(applicationProperties);
+        if (message.getComponentName() != null)
```

Comment:
Let's also skip if the componentName isEmpty. From what I understand of the DTDL the the component name can't be empty, it has to be 1 or more Alphanumeric characters and underscores are allowed after the first character which must be a-z.

```suggestion
        if (message.getComponentName() != null && !message.getComponentName().isEmpty())
```